src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -829,8 +848,11 @@ private function camelize($string)
      *
      * @return array|null An array containing the adder and remover when found, null otherwise
      */
-    private function findAdderAndRemover(\ReflectionClass $reflClass, array $singulars)
+    private function findAdderAndRemover(\ReflectionClass $reflClass, array $singulars, PropertyMetadata $metadata = null)
     {
+        if ($metadata && $metadata->getAdder() && $metadata->getRemover()) {
```

Comment:
I couldn't find anything in this PR related to this requirement, it should be clear somewhere that in one is set this other is required.
Although I feel like one should should be enough, perhaps allowing to mix adder and remover guessing?